### DISCUSSION TITLE
Picking library for Lyon shapes?

### DISCUSSION THREAD

**Discussion by SUPERCILEX** [bevyengine/bevy#4598] [Q&A]
I'm trying to support picking lyon shapes (and ideally also snapping them to a grid, but that's kind of seperate), but am not sure how to go about it. `bevy_mod_picking` doesn't seem to work with Lyon.

**Comment by Nilirad** [comment]
I am assuming that by “Lyon” you are actually referring to `bevy_prototype_lyon`.

If I remember correctly, `bevy_mod_picking` does not work with `bevy_prototype_lyon` shapes, because the meshes used for drawing the shapes were different from the ordinary 3D ones. However, this may now change since it now uses `Mesh2D`, a built in Bevy type. When `bevy_mod_picking` will be able to pick `Mesh2D`, it should also be able to pick shapes created with `bevy_prototype_lyon`.

**Reply by SUPERCILEX** [reply]
Makes sense, thanks! Filed: https://github.com/aevyrie/bevy_mod_picking/issues/130

**Comment by aevyrie** [comment]
This is now solved by https://github.com/aevyrie/bevy_mod_picking/pull/133
